### DISCUSSION TITLE
preview subshell doesn't load any functions

### DISCUSSION THREAD

**Discussion by drfiresign** [PatrickF1/fzf.fish#246] [Q&A] [answered]
Hi, I'm sure there's something happening that's unique to my setup but I thought I'd ask just to be sure. I'm on MacOS 12.4 using fish, version 3.5.0. I've run `fisher update` before trying any of this and the only plugins I have installed are `fisher` and `fzf.fish`.

When I use the shortcut to browse files, I get the file list as expected with the preview window on the right. However, in the preview window it says 
```txt
fish: Unknown command: _fzf_preview_file
fish:
_fzf_preview_file 'whatever filename the cursor is on'
^
```

This also occurs with any of the functions that would be used to display a file in a preview window such as `_fzf_extract_var_info`:
```txt
fish: Unknown command: _fzf_extract_var_info
fish:
_fzf_extract_var_info 'VARIABLE_NAME' /var/folders/etc/etc/etc
^
```

I've confirmed that in my top level shell the `_fzf_preview_file` function works. I can run that with a filename and it will open up `bat` with the expected file contents. I've also confirmed this occurs in both Apple's terminal.app and Alacritty.

Most of my configuration files are actually symlinked into `~/.config/` from my dotfiles git repo in another location on my system. Maybe the subshell that's being opened won't load symlink configuration files? That's the only thing I can think of here that's impacting things, but I've never encountered a problem with this setup before 🤷🏻 

Any ideas you have would be appreciated, thanks!

**Comment by PatrickF1** [comment]
Does (command --search fish) result in the correct fish installation? https://github.com/PatrickF1/fzf.fish/blob/main/functions/_fzf_wrapper.fish#L5

**Comment by drfiresign** [comment]
It does indeed:
```
$ command --search fish
/usr/local/bin/fish
```

I installed using homebrew so this too is a symlink:
```
$ exa -l (command --search fish)
lrwxr-xr-x 29 drfiresign 21 Jun 11:37 /usr/local/bin/fish -> ../Cellar/fish/3.5.0/bin/fish
```

**Comment by PatrickF1** [comment]
Sorry this is out of my expertise. Maybe try asking in fish-shell issues why a newly spawned fish session can't autoload those files?

**Comment by drfiresign** [comment]
No problem, thanks for taking a look at it!

**Comment by drfiresign** [comment, marked as answer]
I think I worked this out. I had file in one of my `~/.config/` directories which also had a corresponding file in my dotfiles repo. So my `stow -R config` command that I run after adding a new file to the dotfiles kept failing and wasn't picking up the changes I was making to the loading order of the files in `~/.config/fish/functions`. That was causing the load failure. 

Thanks for your help 👍🏻
